test/fixtures/linter/rules/CSPCompliance/NoInlineJS.html

```diff
@@ -0,0 +1,39 @@
+<html>
+
+<head>
+	<title>No inline JS</title>
```

Comment:
From my discussion with FW, let's add a meta tag to see whether the TS compiler still handles the document fine.

Note how the meta tag does not close. That's valid HTML but not valid XML (hence not valid JSX? 🤔)

Reply:
<meta charset="UTF-8"> is indeed invalid, but 
<meta charset="UTF-8**/**> is a valid one

Reply:
If this works, shouldn't we then see some findings in the snapshot result? (e.g. usage of global "sap.ui.model.json.JSONModel")

Reply:
> BTW: https://facebook.github.io/jsx/#sec-jsx suggests that JSX is really XML-oriented, not HTML5. The syntax does not ~foresee~ allow HTML5 void tags like `<meta>`, `<br>` or `<img>`.

Please check my [comment above](https://github.com/SAP/ui5-linter/pull/48#discussion_r1542879246)
It actually is able to do some of the checks, especially the global ones. As far as I get it, it misses the transpilation part that we do prior analyzing files and that's why it's unable to detect deprecated dependencies

Reply:
So why are no findings is reported for this test fixture? Can we enhance it to cover some of the findings that are working?

Reply:
Good point! I suppose it's because of the transpilation phase that's missing here. The access to global variables is wrapped in `sap.ui.controller`. I wonder why `sap.ui.controller` is not reported. Do we make exceptions for certain cases?

However, if you put the `sap.ui.model.json.JSONModel()` directly into the script tag, it's reported correctly.
I'll debug and will create a tests for those cases.

Reply:
I think I got it how it works. The HTML tags are treated as JSX code and respectively converted into NodeObjects. Their contents are treated as text, but depending on how this text is split into chunks, **there's a chance** (but not necessarily) this text also to be parsed and as it is in our case to report issues.
However, this seems very fragile and does not produce constant results.

Given that sometimes it would produce an expected report and sometimes not, for similar inputs I'd suggest disabling this feature and handling that case in the follow-up backlog.

Any objections?